### PR TITLE
Make timeout configurable and finish custom headers

### LINKED ISSUE #12: Unable to configure the window parameter

### PR DESCRIPTION
**Pull request by brittcyr**
Fixes #12

### CODE REVIEW

**Review by meyafex — APPROVED** [review]
Good job guys!

File: backpack_client.ts [hunk 1]
```diff
@@ -189,12 +189,21 @@ const rawRequest = async (
...
-
-  constructor(privateKey: string, publicKey: string) {
-    this.config = { privateKey, publicKey };
+  private config: any;
+  private customHeaders: object;
+
+  constructor(
+    privateKey: string,
+    publicKey: string,
+    customHeaders: object = {},
```
Comment (sndmndss) [inline backpack_client.ts:202]: customHeaders: Record<string, string> = {},

File: backpack_client.ts [hunk 2]
```diff
@@ -189,12 +189,21 @@ const rawRequest = async (
...
+ * @param {object}        customHeaders custom headers for all requests
+ * @param {number}        windowMs timeout window for all requests in millis
  */
 export class BackpackClient {
-  public config: any;
-
-  constructor(privateKey: string, publicKey: string) {
-    this.config = { privateKey, publicKey };
+  private config: any;
+  private customHeaders: object;
```
Comment (sndmndss) [inline backpack_client.ts:197]: private customHeaders: Record<string, string>;

File: backpack_client.ts [hunk 3]
```diff
@@ -189,12 +189,21 @@ const rawRequest = async (
...
  * @param {string}        publicKey  base64 encoded
+ * @param {object}        customHeaders custom headers for all requests
+ * @param {number}        windowMs timeout window for all requests in millis
  */
 export class BackpackClient {
-  public config: any;
-
-  constructor(privateKey: string, publicKey: string) {
-    this.config = { privateKey, publicKey };
+  private config: any;
```
Comment (sndmndss) [inline backpack_client.ts:196]: private config: {
    privateKey: string;
    publicKey: string;
    timeout: number;
  };

**Review by DonDuala — APPROVED** [review]

**Pull request merged**